### PR TITLE
Improvising qos tests by tunning the qos params for single_asic, single_dut_multi_asic and  multi_dut

### PR DESCRIPTION
**Pull request by ansrajpu-git**
### Description of PR
This PR is in continuation of PR# https://github.com/sonic-net/sonic-mgmt/pull/8149
which was originally part of PR# https://github.com/sonic-net/sonic-mgmt/pull/6946

The existing QoS (test_qos_sai.py) is written to accommodate a single asic on a single Dut. But, we require the same tests to be executed against a T2 chassis (with single/multi-asic linecards) and multi-asic pizza boxes.

Summary:
 This PR is in continuation to above PR's mentioned

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
1.Qos test cases failed with intermittent errors
#### How did you do it?
Two issues are addressed here :

1.The dscp queue mapping for LossyQueue Test changed in config file to map to queue 1 of traffic-class instead of  0  since disabling the tx and filling up the queue 0  prevents the lacp packets going out and port channel goes down

2.During Qos test on transmission disable and enable, sometimes on test failure the port dangles in a transmission disable state and did not recover. Switching the step to enable the transmission port before the BCMSAI credit-watchdog enable , eradicate the test failure due to bad port state.  
#### How did you verify/test it?

Executed qos testcases  on for single_asic ,single_dut_multi_asic & multi_dut

### CODE REVIEW

**Comment by xwjiang-ms** [comment]
pre-commit check is a mandatory check, please fix detected issues

**Comment by vmittal-msft** [comment]
This PR is dependent on https://github.com/sonic-net/sonic-mgmt/pull/8149. Once we have #8149 pushed in, we can push in this as well.

**Comment by judyjoseph** [comment]
@vmittal-msft Could you check and review this PR ?

File: tests/qos/files/qos_params.j2c.yaml [hunk 1]
```diff
@@ -51,11 +51,11 @@ qos_params:
                     pkts_num_dismiss_pfc: 200
                     pkts_num_margin: 150
                 lossy_queue_1:
-                    dscp: 8
+                    dscp: 7
                     ecn: 1
-                    pg: 0
+                    pg: 1
```
Comment (vmittal-msft) [inline tests/qos/files/qos_params.j2c.yaml:56]: This will be fixed by BRCM CSP.

File: tests/saitests/py3/sai_base_test.py [hunk 2]
```diff
@@ -177,8 +178,6 @@ def sai_thrift_port_tx_enable(self, client, asic_type, port_list, target='dst',
             assert 'Success rv = 0' in stdOut[1], "enable wd failed '{}' on asic '{}' on '{}'".format(
                     cmd, self.src_asic_index, self.src_server_ip)
 
-        sai_thrift_port_tx_enable(client, asic_type, port_list, target=target)
-
     def sai_thrift_port_tx_disable(self, client, asic_type, port_list, target='dst'):
         if self.platform_asic and self.platform_asic == "broadcom-dnx":
             # need to enable watchdog on the source asic using cint script
```
Comment (vmittal-msft) [inline tests/saitests/py3/sai_base_test.py:183]: Please check if there is any restriction on when credit-watchdog can be enabled/disabled in terms of traffic in the system.

File: tests/saitests/py3/sai_qos_tests.py [hunk 3]
```diff
@@ -351,6 +351,7 @@ def runTest(self):
         dual_tor = self.test_params.get('dual_tor', None)
         leaf_downstream = self.test_params.get('leaf_downstream', None)
         asic_type = self.test_params['sonic_asic_type']
+        platform_asic = self.test_params['platform_asic']
```
Comment (vmittal-msft) [inline tests/saitests/py3/sai_qos_tests.py:354]: Please revert as we are fixing the queueu swap issue.

File: tests/qos/files/qos_params.j2c.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@ qos_params:
     j2c+:
         topo-any:
             100000_300m:
-                pkts_num_leak_out: 51
+                pkts_num_leak_out: 5
```
[the remark this answers is not in the document]
Comment (ansrajpu-git) [reply tests/qos/files/qos_params.j2c.yaml:5]: this change is to fine tune the parameter to get consistent success results.

File: tests/saitests/py3/sai_qos_tests.py (hunk 3 above)
Comment (ansrajpu-git) [reply tests/saitests/py3/sai_qos_tests.py]: Reverted the commit for DscpQueueMapping test

File: tests/qos/files/qos_params.j2c.yaml [hunk 5]
```diff
@@ -157,9 +157,9 @@ qos_params:
                     pkts_num_dismiss_pfc: 200
                     pkts_num_margin: 150
                 lossy_queue_1:
-                    dscp: 8
+                    dscp: 7
                     ecn: 1
-                    pg: 0
```
Comment (vmittal-msft) [inline tests/qos/files/qos_params.j2c.yaml:162]: please keep existing values as we will have fix for this.

File: tests/qos/files/qos_params.j2c.yaml (hunk 4 above)
Comment (vmittal-msft) [inline tests/qos/files/qos_params.j2c.yaml:5]: Since leakout is used at multiple places in the tests, did we do complete run to make sure no other test is broken ?

File: tests/saitests/py3/sai_qos_tests.py [hunk 6]
```diff
@@ -456,7 +457,11 @@ def runTest(self):
             # queue 3/4  1                 1               1                                                1                                         1                         # noqa E501
             # queue 5    1                 1               1                                                1                                         1                         # noqa E501
             # queue 7    0                 1               1                                                1                                         1                         # noqa E501
-            assert (queue_results[QUEUE_0] == 1 + queue_results_base[QUEUE_0])
+            if platform_asic and platform_asic == "broadcom-dnx":
```
Comment (vmittal-msft) [inline tests/saitests/py3/sai_qos_tests.py:460]: Shall we revert them as fix from BRCM should fix this ?

File: tests/saitests/py3/sai_qos_tests.py [hunk 7]
```diff
@@ -3082,6 +3087,15 @@ def runTest(self):
                 assert (fill_leakout_plus_one(self, src_port_id, dst_port_id,
                                               pkt, int(self.test_params['pg']), asic_type))
 
+            # send packets short of triggering egress drop
```
Comment (vmittal-msft) [inline tests/saitests/py3/sai_qos_tests.py:3090]: why do we need this ? Also, Can we have this change applicable to chassis only ?

File: tests/saitests/py3/sai_qos_tests.py [hunk 8]
```diff
@@ -3111,7 +3125,8 @@ def runTest(self):
             assert (recv_counters[pg] == recv_counters_base[pg])
             # recv port no ingress drop
             for cntr in ingress_counters:
-                assert (recv_counters[cntr] == recv_counters_base[cntr])
+                if platform_asic and platform_asic == "broadcom-dnx" and cntr == 1:
+                    assert(recv_counters[cntr] == recv_counters_base[cntr])
```
Comment (vmittal-msft) [inline tests/saitests/py3/sai_qos_tests.py:3129]: what is the purpose of this change ? Can we have this applicable to chassis only ?

File: tests/qos/files/qos_params.j2c.yaml (hunk 4 above)
Comment (ansrajpu-git) [reply tests/qos/files/qos_params.j2c.yaml:5]: We have tested same value every weekly build and till now it is working perfectly.

File: tests/saitests/py3/sai_qos_tests.py (hunk 6 above)
Comment (ansrajpu-git) [reply tests/saitests/py3/sai_qos_tests.py]: this commit has been reverted  #43740a5c0. The BRCM fix is provided. Testing is in progress for the fix

File: tests/saitests/py3/sai_qos_tests.py (hunk 8 above)
Comment (saksarav-nokia) [reply tests/saitests/py3/sai_qos_tests.py:3124]: Vineet, If i remember correctly the packet drops are counted as part of queue drop stats and not port drop stats , So the In Discards counter was not working and we had to bypass that  counter check for dnx

File: tests/saitests/py3/sai_qos_tests.py (hunk 7 above)
Comment (ansrajpu-git) [reply tests/saitests/py3/sai_qos_tests.py]: Incorporated the comments. Kindly review

**Review by vmittal-msft — APPROVED** [review]

**Review by vmittal-msft — APPROVED** [review]

**Comment by ansrajpu-git** [comment]
@XuChen-MSFT ,@wsycqyz ,kindly review & approve
